lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp

```diff
@@ -4564,6 +4573,116 @@ void glow::convertQuantizedConstants(Function *F, CompilationContext &cctx) {
   runDCEPass(F, cctx);
 }
 
+// External quantization pass, If the backend doesn't support a node in INT8,
+// convert that node to FP32 or FP16 depending upon the convert-to-fp16 flag.
+// Along with data type modification, add dequantization and quantization nodes
+// respectively at the input and the output of the target node.
+void glow::externalQuantizationPass(const Backend &B, Function *F,
+                                    CompilationContext &cctx) {
+
+  const PrecisionConfiguration &precConfig = cctx.precisionConfig;
+
+  for (Node &node : F->getNodes()) {
+    bool isFP16Supported = false;
+    bool isSupported = B.isOpSupported(node);
+
+    // check if all of the outputs of node are quantized type
+    bool isAllResQuantized = true;
+    for (size_t i = 0, e = node.getNumResults(); i < e; i++) {
+      if (node.getType(i)->isFPType()) {
+        isAllResQuantized = false;
+        break;
+      }
+    }
+
+    if (!isSupported && isAllResQuantized) {
```

Comment:
nit: Remove level of indentation
```suggestion
    if (isSupported || !isAllResQuantized) {
      continue;
    }
```